content/components/data-table.mdx

```diff
@@ -0,0 +1,604 @@
+
+<Box as="p" flex={1}>
+  The column header starts unsorted. The first click sorts in ascending order, the second click sorts in descending
+  order, and the third click removes the sort. When the sort is removed, the rows should be in the order they were in
+  when the table was first rendered
+</Box>
+
+<CustomVideoPlayer
+  width="456"
+  loop
+  src="https://user-images.githubusercontent.com/2313998/205755293-c03ee5f6-faa8-4f73-89e5-4bf33cc06086.mp4"
+/>
+</Box>
+
+#### States
+
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} sx={{gap: 3}}>
+
+<Box as="ul" flex={1}>
+  <li>Unsorted (default)</li>
+  <li>Ascending sort active</li>
+  <li>Descending sort active</li>
+  <li>Sort in progress</li>
+</Box>
+
+<img
+  width="456"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/205361240-85add374-1599-48bf-b6f0-c8e958fbd2dc.png"
+/>
+</Box>
+
+### Filtering
+
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} sx={{gap: 3}}>
+
+<Box as="p" flex={1}>
+  The "query builder" pattern is used to show rows that match the user's query.
+</Box>
+
+<img
+  width="456"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/205361241-318146a6-d2ff-47e7-96fc-4f1bd4cd5698.png"
+/>
+</Box>
+
+## Accessibility
+
+### Labeling and describing the table
+
+The table must have a title, and can optionally have a subtitle to describe more context. If you encounter a case where there is enough context that the table doesn’t need a visible label, you may hide it so it’s only accessible to assistive technologies.
+
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} mb={3} sx={{gap: 3}}>
+
+<Box as="p" flex={1}>
+  If you use the table header, the table's title and subtitle will automatically be associated with the table for
+  assistive technologies.
+</Box>
+
+<img
+  width="456"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/205361243-62d0b497-a4ba-44b5-8e3e-258cdd6e53e1.png"
+/>
+</Box>
+
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} sx={{gap: 3}}>
+
+<Box as="p" flex={1}>
+  If your table's title exists somewhere else on the page (for example, in the PageHeader), then you must manually
+  associate that contextual information with the table using ARIA.
+</Box>
+
+<img
+  width="456"
+  role="presentation"
+  src="https://user-images.githubusercontent.com/2313998/205361244-8a1f527a-e1f3-45a3-ab13-5fdf82476038.png"
+/>
+</Box>
+
+### Labeling actions
+
+<Box display="flex" flexDirection="column" sx={{gap: 3}}>
+<Box display="flex" flexDirection={['column', 'column', 'column', 'column', 'row']} sx={{gap: 3}}>
+
+<Box as="p" flex={1}>
+  The row header string will be prepended to the ARIA text of action buttons. For examples, Download {`{row header}`}“,
+  “Actions:{`{row header}`}”
+</Box>
```

Comment:
This seems like it could add extra verbosity that a screen reader user might not expect and couldn't control with their verbosity settings. Something that could be confusing when navigating between rows with this labeling technique is that the row header could end up being announced twice.

If I understand right, navigating tables with NVDA/JAWS/VoiceOver works like:

- If it supports announcing a cell, it will read both the column and row headers before the cell content
- When changing columns, it will announce the column header before the cell content
- When changing rows, it will announce the row header (if it exists)

Tried to confirm this over in a [Loom video](https://www.loom.com/share/6d7996881cfb45d9b7aa320c9104f2fe). Let me know if I tested anything incorrectly!

Would it make sense for the row action to have its semantic label and rely on the markup semantics of the row header to add the context for the end user? Would love to hear your thoughts here

Reply:
Yea, I think it makes sense to let the column header and row header semantics add context for the end user.

cc @ericwbailey - what do you think of that approach?

Reply:
I like it! 

My one concern is this might turn it into a content authoring issue that we can't anticipate, but I don't think that should block this progress.

Also thank you for the Loom video, @joshblack! That was super helpful 🎉